Source/FieldSolver/FiniteDifferenceSolver/FiniteDifferenceAlgorithms/CKCAlgorithm.H

```diff
@@ -0,0 +1,229 @@
+/* Copyright 2020 Remi Lehe
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#ifndef WARPX_FINITE_DIFFERENCE_ALGORITHM_CKC_H_
+#define WARPX_FINITE_DIFFERENCE_ALGORITHM_CKC_H_
+
+#include <AMReX_REAL.H>
+#include <AMReX_Array4.H>
+#include <AMReX_Gpu.H>
+
+struct CKCAlgorithm {
+
+    static void InitializeStencilCoefficients(
+        std::array<amrex::Real,3>& cell_size,
+        amrex::Gpu::ManagedVector<amrex::Real>& stencil_coefs_x,
+        amrex::Gpu::ManagedVector<amrex::Real>& stencil_coefs_y,
+        amrex::Gpu::ManagedVector<amrex::Real>& stencil_coefs_z ) {
+
+        using namespace amrex;
+
+        // Compute Cole-Karkkainen-Cowan coefficients according
+        // to Cowan - PRST-AB 16, 041303 (2013)
+        Real alphax=0, alphay=0, alphaz=0;
+        Real betaxy=0, betaxz=0, betayx=0, betayz=0, betazx=0, betazy=0;
+        Real gammax=0, gammay=0, gammaz=0;
+        Real const inv_dx = 1./cell_size[0];
+        Real const inv_dy = 1./cell_size[1];
+        Real const inv_dz = 1./cell_size[2];
+#        if defined WARPX_DIM_3D
+            const Real delta = std::max( { inv_dx,inv_dy,inv_dz } );
+            const Real rx = (inv_dx/delta)*(inv_dx/delta);
+            const Real ry = (inv_dy/delta)*(inv_dy/delta);
+            const Real rz = (inv_dz/delta)*(inv_dz/delta);
+            const Real beta = 0.125*(1. - rx*ry*rz/(ry*rz + rz*rx + rx*ry));
+            betaxy = ry*beta;
+            betaxz = rz*beta;
+            betayx = rx*beta;
+            betayz = rz*beta;
+            betazx = rx*beta;
+            betazy = ry*beta;
+            gammax = ry*rz*(0.0625 - 0.125*ry*rz/(ry*rz + rz*rx + rx*ry));
+            gammay = rx*rz*(0.0625 - 0.125*rx*rz/(ry*rz + rz*rx + rx*ry));
+            gammaz = rx*ry*(0.0625 - 0.125*rx*ry/(ry*rz + rz*rx + rx*ry));
+            alphax = 1. - 2.*betaxy - 2.*betaxz - 4.*gammax;
+            alphay = 1. - 2.*betayx - 2.*betayz - 4.*gammay;
+            alphaz = 1. - 2.*betazx - 2.*betazy - 4.*gammaz;
+            betaxy *= inv_dx;
+            betaxz *= inv_dx;
```

Comment:
could move up, const them in place.

Reply:
I think I agree with this comment, we could probably replace
```c++
            Real alphax=0;
            Real betaxy=0;
            Real gammax=0;
            betaxy = ry*beta;
            gammax = ry*rz*(0.0625 - 0.125*ry*rz/(ry*rz + rz*rx + rx*ry));
            alphax = 1. - 2.*betaxy - 2.*betaxz - 4.*gammax;
            betaxy *= inv_dx;
            alphax *= inv_dx;
```
by
```c++
            Real const betaxy = ry*beta*inv_dx;
            Real const gammax = ry*rz*(0.0625 - 0.125*ry*rz/(ry*rz + rz*rx + rx*ry))*inv_dx;
            Real const alphax = (1. - 2.*ry*beta - 2.*rz*beta - 4.*gammax*dx ) * inv_dx;
```
a bit more verbose but `const`-friendly.

Reply:
Yes, but even better with `_rt`:
```c++
            Real const gammax = ry*rz*(0.0625_rt - 0.125_rt*ry*rz/(ry*rz + rz*rx + rx*ry))*inv_dx;
            Real const alphax = (1._rt - 2._rt*ry*beta - 2._rt*rz*beta - 4.*gammax*dx ) * inv_dx;
```